airbyte-integrations/connectors/source-mssql/src/test-integration/java/io/airbyte/integrations/source/mssql/MssqlSourceComprehensiveTest.java

```diff
@@ -0,0 +1,402 @@
+  protected Database setupDatabase() throws Exception {
+    container = new MSSQLServerContainer<>("mcr.microsoft.com/mssql/server:2019-latest")
+        .acceptLicense();
+    container.start();
+
+    final JsonNode configWithoutDbName = Jsons.jsonNode(ImmutableMap.builder()
+        .put("host", container.getHost())
+        .put("port", container.getFirstMappedPort())
+        .put("username", container.getUsername())
+        .put("password", container.getPassword())
+        .build());
+
+    final Database database = getDatabase(configWithoutDbName);
+    database.query(ctx -> {
+      ctx.fetch(String.format("CREATE DATABASE %s;", DB_NAME));
+      ctx.fetch(String.format("USE %s;", DB_NAME));
+      return null;
+    });
+
+    config = Jsons.clone(configWithoutDbName);
+    ((ObjectNode) config).put("database", DB_NAME);
+
+    return database;
+  }
+
+  private static Database getDatabase(JsonNode config) {
+    return Databases.createDatabase(
+        config.get("username").asText(),
+        config.get("password").asText(),
+        String.format("jdbc:sqlserver://%s:%s",
+            config.get("host").asText(),
+            config.get("port").asInt()),
+        "com.microsoft.sqlserver.jdbc.SQLServerDriver",
+        null);
+  }
+
+  @Override
+  protected String getNameSpace() {
+    return "dbo";
+  }
+
+  @Override
+  protected String getImageName() {
+    return "airbyte/source-mssql:dev";
+  }
+
+  @Override
+  protected JsonNode getConfig() {
+    return config;
+  }
+
+  @Override
+  protected void tearDown(TestDestinationEnv testEnv) throws Exception {
+    container.stop();
+    container.close();
+  }
+
+  @Override
+  protected void initTests() {
+    addDataTypeTestData(
+        TestDataHolder.builder()
+            .sourceType("bigint")
+            .airbyteType(JsonSchemaPrimitive.NUMBER)
+            .addInsertValues("-9223372036854775808", "9223372036854775807", "0", "null")
+            .addExpectedValues("-9223372036854775808", "9223372036854775807", "0", null)
+            .build());
+
+    addDataTypeTestData(
+        TestDataHolder.builder()
+            .sourceType("int")
+            .airbyteType(JsonSchemaPrimitive.NUMBER)
+            .addInsertValues("null", "-2147483648", "2147483647")
+            .addExpectedValues(null, "-2147483648", "2147483647")
+            .build());
+
+    addDataTypeTestData(
+        TestDataHolder.builder()
+            .sourceType("smallint")
+            .airbyteType(JsonSchemaPrimitive.NUMBER)
+            .addInsertValues("null", "-32768", "32767")
+            .addExpectedValues(null, "-32768", "32767")
+            .build());
+
+    addDataTypeTestData(
+        TestDataHolder.builder()
+            .sourceType("tinyint")
+            .airbyteType(JsonSchemaPrimitive.NUMBER)
+            .addInsertValues("null", "0", "255")
+            .addExpectedValues(null, "0", "255")
+            .build());
+
+    // TODO BUG Returns values true, false instead of 1 and 0.
```

Comment:
I think bit is used to express boolean in MSSQL so I think this is behaving as expected

https://stackoverflow.com/questions/1777257/how-do-you-create-a-yes-no-boolean-field-in-sql-server

Reply:
An official guide (https://docs.microsoft.com/en-us/sql/t-sql/data-types/bit-transact-sql?view=sql-server-ver15) says that the server converts true to 1 when you save it, but nothing about that stored 1 will be converted to true, that's why I considered it as a bug. Updated tests. made these 'true\false' as an expected behavior as you said. Thanks

Reply:
It looks like `bit` columns are returned as `true` / `false` if there is no explicit conversion:

https://dbfiddle.uk/?rdbms=sqlserver_2019&fiddle=a2f618a30611ba2e995219b5fa24766d

<img width="849" alt="Screen Shot 2021-06-25 at 07 47 10" src="https://user-images.githubusercontent.com/1933157/123442267-901d3080-d589-11eb-9e8b-dd319b7e8f95.png">